rfd/0103-application-access-web-ui-auth-flow.md

```diff
@@ -12,8 +12,7 @@ application through Teleport Web UI.
 
 ## Why
 
-The method used to set cookies on a different domain requires a few changes to Teleport's default security
-mechanisms. It's important these changes are documented and understood for future reference.
+To document and understand the HTTP application access authn flow for future reference.

```

Comment:
I'm still a little fuzzy on what's going on here.

Before proxying a user's request to an app, teleport has to:

1. verify the user is authorized
2. create an app session?
3. anything else?

Which of the above steps is this RFD describing?

If it's 1, I assume there must be a good reason why we don't just derive the user's identity from their existing bearer token, check RBAC, and proxy the request.

Can you help me understand why this extra machinery is necessary?

Reply:
> there must be a good reason why we don't just derive the user's identity from their existing bearer token, check RBAC, and proxy the request.

Sorry I wasn't too sure what you meant by ^. I edited the sequence diagram to include logging in. Does that make more sense?

I'll try to clarify

> Verify the user is authorized

User has to be authenticated for the app access flow to succeed.

The web launcher route in the web UI is in between the auth checker, so web UI will check if user is authenticated, if not, user is routed to login (this part was missing in diagram). 

In order to [create an app session](https://github.com/gravitational/teleport/blob/ebf69b6a6bf74db668b30f93e50200c692e996fe/lib/web/apiserver.go#L610) during app access flow, user has to be authenticated (it's a protected endpoint)

> Create an app session

When forwarding app request to app services, we authenticate the request by looking up existing app session (provided by session cookie) and checking for equality of the fetched sessions bearer token with the one passed by subject cookie. The app access flow sets these required app cookies